keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlArtifactRepository.kt

```diff
@@ -215,6 +221,84 @@ class SqlArtifactRepository(
       .execute()
   }
 
+  override fun versionsByEnvironment(deliveryConfig: DeliveryConfig): List<EnvironmentArtifactsSummary> {
```

Comment:
Can we change the method signature to support limiting? For `versionsInEnvironment`, it would probably be ok to `order by ENVIRONMENT_ARTIFACT_VERSIONS.APPROVED_AT desc` and take the topN.

Reply:
Could we just default to a limit? I worry that it seems misleading to specify a limit in the signature that we may blow because it's the product of multiple queries. I don't think there's any reliable way to sort the 2nd query in SQL.